src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -156,6 +222,673 @@ MultiRouterPlanCreate(Query *originalQuery, Query *query,
 
 
 /*
+ * Creates a router plan for INSERT ... SELECT queries which could consists of
+ * multiple tasks.
+ *
+ * The function never returns NULL, it errors out if cannot create the multi plan.
+ */
+static MultiPlan *
+CreateMultiTaskRouterPlan(Query *originalQuery,
```

Comment:
Yes, that's right. My motivation was that we could implement multiple shard select modify queries here as well.

Should I rename to a more specific name for INSERT ... SELECT?